runner/pool/pool.go

```diff
@@ -589,7 +589,15 @@ func (r *basePoolManager) cleanupOrphanedGithubRunners(runners []*github.Runner)
 			slog.DebugContext(
 				r.ctx, "updating instances cache for pool",
 				"pool_id", pool.ID)
-			poolInstances, err = provider.ListInstances(r.ctx, pool.ID)
+			listInstancesParams := common.ListInstancesParams{
+				ListInstancesV011: common.ListInstancesV011Params{
+					ProviderBaseParams: common.ProviderBaseParams{
```

Comment:
At this point, the base params will always contain the same thing. Please create a function in the `basePoolManager{}` that returns the `ProviderBaseParams{}` struct. And you can just call it.

The function should be protected by `r.mux.Lock()` as the pool info can be changed and updated by the DB watcher.

Reply:
We might also want to switch on the interface version in the future and return the appropriate `ListInstancesParams{}`